### ISSUE TITLE
Disallow disabling `_field_names`

### ISSUE THREAD

**Issue by jpountz** [elastic/elasticsearch#27239]
Now that `_field_names` has basically no overhead anymore (#26930), we should ignore the `enabled` setting in 6.x (and log deprecation) and reject it in 7.x, or maybe 8,x if we want to maintain compatibility for templates.

cc @colings86

**Comment by colings86** [comment]
Caveat to this: `_fieldNames` has basically no overhead in the case where doc_values and norms are enabled. If doc_values are disabled then `_field_names` will have the same overhead as before.

**Comment by javanna** [comment]
@elastic/es-search-aggs

**Comment by ppf2** [comment]
As part of this ticket, can we also clarify https://www.elastic.co/guide/en/elasticsearch/reference/6.2/tune-for-indexing-speed.html#_disable_literal__field_names_literal here to mention that there is much less overhead now for most use cases except for the caveats mentioned above?  Thx!

**Comment by jimczi** [comment]
We discussed this issue internally and agreed that this is something that we still want to do.
Since it's not high priority we also said that we should at least clarify the documentation to mention the new cost of the `_field_names` field so I am adding the `>docs` and `adoptme` label.

**Comment by cbuescher** [comment]
I just checked the docs that @ppf2 linked to above and the section about disabling `_field_names` has been removed from the "Tune for indexing speed" page as of https://github.com/elastic/elasticsearch/pull/37717.
I will check if there are other locations in the docs where it would make sense to mention the cost of `_field_names`. That should cover the docs side of things for now, we would still need to deprecate the `enabled` setting separately.
